### DISCUSSION TITLE
My AdGuard Home is not working with all devies

### DISCUSSION THREAD

**Discussion by libtech4** [AdguardTeam/AdGuardHome#5473] [Q&A]
Hi Github community, 
It's frustrating that I spent almost $200 to get Raspberry Pi just for AdGuard Home so that I can block ads, most importantly content from my network so that my kids won't get any inappropriate content while using the internet. It's not working with apple devices and chrome book. 
I believe I followed all the steps. Let me explain what I did maybe I missed something. 
1- I opened 53, 80, and 3000 firewall ports
2- I ran this command since I was not sure what to download. **sudo curl -s -S -L https://raw.githubusercontent.com/AdguardTeam/AdGuardHome/master/scripts/install.sh | sh -s -- -v** it installed the AdGuard and it display my Pi' IP (192.168.0.20:3000) with port to log in to UI.  
3- I typed ip with port and created an account.
4- From the General setting I checked the "Use AdGuard parental control web service" and the "Use safe search" option.
5- From DNS Setting added the below three addresses and checked "Enable DNSSEC"
https://dns.cloudflare.com/dns-query
https://dns.quad9.net/dns-query
https://dns.google/dns-query
https://family.cloudflare-dns.com/dns-query 
6- from DNS Blocklists I checked all options under the Security and four options 1Host lite and 1 Host mini, AdGuard DNS filter, and AdAway default blocklist.
7- From Block Services I enable the block option for TikTok, OnlyFan, Xbox live chat, and some other services.
8- I added AdGuard address 192.168.0.20 and ipv6 address in my Netgear router. 
9- Unabled and re-enabled network from my devices such as a Linux server, iPhones, iPad, Chromebook, etc. 

These are issues I'm having.
1- On my Linux server I can still see two other DNS in resolve.conf file, the first two are from AdGuard one is ipv4 and the second is v6. The other two are v6 addresses and I can see those in my iPhone's DNS setting as well. When I delete those manually from resolve.conf file and restart the network those come back.
2- Ads are not getting blocked from iPhones and iPad. Though I can see the AdGuard address in the DNS setting and DNS is automatic. When I change to manual and delete the other two addresses and leave the AdGuard one, the internet won't work. Under the wifi name, I can see a yellow message saying "No internet connection".
3- In my amazon firestick under DNS setting it shows the AdGuard address (192.168.0.20), also shows google's DNS 8.8.8.8. There isn't a way to add manual DNS. 
4- Google home stopped working, it shows offline. The only time it'll work is if I changed router ipv6 DNS from manual to automatic.
5-  My garage door app (myQ) goes offline mode.
6-  My daughter's chrome book's internet won't work, Chromebook is from school. Either I have to enter DNS from the isp provider to add google's DNS in her Chromebook's network setting. 
7- Alexa speaker stop working, and I get a network disconnected error from the app.

It's only working with a laptop that I have RHEL8 GUI installed though I can see old DNS entries in resolve.con file. But tiktok is blocked and ads getting blocked. 
Can someone please help to fix the issue? I have already spent almost a month on this and I am helpless and can't figure it out. I uninstalled and reinstalled AdGuard many times. 
Thank you for your help!

**Comment by ishanjain28** [comment]
> 1- On my Linux server I can still see two other DNS in resolve.conf file, the first two are from AdGuard one is ipv4 and the second is v6. The other two are v6 addresses and I can see those in my iPhone's DNS setting as well. When I delete those manually from resolve.conf file and restart the network those come back.

Your DHCP server(the Netgear) device is still advertising those DNS addresses. Please recheck the DNS fields in the DHCP section and make sure only adguard addresses are present in that list. There is a possibility(if this netgear device is owned by your ISP) they are not disabling advertisement of those servers even when you specify your own DNS servers. 


Points 2-7 show, your adguard home instance is not able to resolve any domains. There can be two reasons here. 

1. You have blocked that domain in AGH's settings
2. Adguard home doesn't have internet access or it's not able to reach those configured DNS servers. 

For debugging this I'll suggest,

1. Disable all the block lists, blocked services for now. 
2. Add adguard home's address in the DNS server section on a machine and then try opening a site. If this works, Then you probably disabled too many things that are required for normal operation. 
3. If you still can not open any thing, Please do these two things. 

a.
1. Login using SSH on your adguard home machine.
2. Run `ping -c 3 1.1.1.1`. See if this is getting normal responses from cloudflare.
3. Run `dig +short @192.168.0.20 google.com`. See if this gets any IP addresses in response 

b. 
1. Open adguard home->Query Log and check if you are getting any requests from devices in your LAN. 
2. If you are getting requests, Check if it resolved successfully or failed or what happened to it?
3. If you don't see any requests from other devices in your LAN, Go to any other device on your LAN and try to open the adguard home interface(or ping the machine running adguard home, `ping -c 3 192.168.0.20`). If it's a Linux/macos device, Try to resolve a host name using adguard home by running, `dig +short @192.168.0.20 google.com`. If the domain doesn't resolve or if you can't reach the adguard home machine, then you may have a firewall active on your router or the device running adguard home OR it's just some other networking issue(Configuration issues/some other connectivity issue)

**Reply by libtech4** [reply]
Hi @ishanjain28, sorry I mentioned IP is .20, it's.21, I didn't remember at the time of the post.
1. The modem/router doesn't belong to my ISP provider, I purchased it myself. 
2. Looks like the router is using AdGurard's DNS. I'm attaching IPv4 and v6 screenshots.
3. I unblocked all services.
4. I get an unreachable response when I try to ping 1.1.1.1 or a time-out response when using the dig +short command. Attaching screenshot.
5. Adguard's query log I can see android tablet requests, amazon firestick requests, and my Linux server requests.
6.AdGuard is installed in Raspberry Pi, which I noticed in resolve.conf file that it's showing AdGuard machine IPv4 and v6, as well as the ISP provider's DNS. I'm not sure it should show all of these. I haven't added any addresses manually. Attaching screenshot.
7. I am able to ping AdGuard's server from my other server. 
Any idea how to fix the issue? Thank you for your help.
![AdGuard_Setting](https://user-images.githubusercontent.com/51037316/218292245-b22862a4-7d45-473e-9307-89ca852e00ee.png)
![dig_google](https://user-images.githubusercontent.com/51037316/218292247-c874c791-ef7e-481f-9ff2-26c2583e9bec.png)
![Netgear_IPv6_DNS](https://user-images.githubusercontent.com/51037316/218292249-f7b79a1d-1ddd-49d9-a404-7a328977006f.png)
![Netgear_IPV4_DNS](https://user-images.githubusercontent.com/51037316/218292251-c410653b-6575-4b83-8b91-d5cb39e1dea6.png)
![Ping_1 1 1 1](https://user-images.githubusercontent.com/51037316/218292253-5a49efad-7560-46f8-9f24-edcf46f87113.png)
![Resolved conf](https://user-images.githubusercontent.com/51037316/218292254-bea16248-157c-4847-b03c-e36044b314fd.png)
![Screenshot from 2023-02-11 20-54-24](https://user-images.githubusercontent.com/51037316/218292255-21669d5a-ca3a-4805-a0a4-c19f127e2b91.png)
![Server Ping](https://user-images.githubusercontent.com/51037316/218292256-3c3e9927-99cf-4bf9-9e85-50ee0eb1b5f2.png)

**Comment by ishanjain28** [comment]
If `ping 1.1.1.1` or that dig command are failing then that shows your adguard home instance does not have internet connectivity. T

Try running `ping 1.1.1.1` on other windows/mac/linux devices. If you get the same error on those devices then you just don't have internet access. This can be either because of an outage at your ISP or a configuration issue in your router.

**Reply by libtech4** [reply]
I updated the DNS setting from my router from "Use These DNS Servers" to "Get Automatically from ISP" and rebooted Pi, the internet started working. Then I added AdGuard's address manually on my laptop and iPhone and it's blocking the ads. Updating the DNS of each device manually isn't a good idea. Also, some devices won't have a manual DNS option such as Amazone FireStick and one of my kid's LG tabs. 
When I was using AdGuard's address as DNS in my router, I noticed it added the same DNS to its server. I mean AdGuard server address is 192.168.0.21 and when I checked the server's resolve.conf file the first nameserver was 192.168.0.21. When I rebooted the server it was gone and now I can see ISP's DNS address in the resolve.conf file.
How can make it work with my router I add the address in the router and I don't have to change DNS in every single device. Is it me doing something wrong?

**Reply by libtech4** [reply]
The issue is whenever I replace my router's DNS with AdGuard's address, it also updates the AdGuard server's DNS. Is there a way to hardcode the AdGuard server's DNS so that it won't change if I update the router's address?

**Reply by libtech4** [reply]
I figured out the issue. I wanted to post if someone is a newbie like me and has the same problem. I didn't know the 53/UDP port needed to be open. I opened 53 TCP ports but UDP needed to be open. I just added a DNS service of ports in the firewall it solved the issue. 
The only issue I'm having with ipv6. There isn't an option in the Netgear router to disable ipv6 and adding adguard's ipv6 in the router isn't working. Whenever I replace ISP's ipv6 DNS with adguard's DNS network stops working. Though the ipv6 service is already opened from the firewall. If I could disable ipv6 from my router I couldn't care but the issue is apple devices are ipv6's DNS with adguard's DNS.
